docs/src/tutorials/algorithms/tsp_lazy_constraints.jl

```diff
@@ -0,0 +1,326 @@
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,       #src
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE    #src
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER         #src
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,  #src
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE  #src
+# SOFTWARE.     
+
+# # [Traveling Salesperson Problem (via callbacks)](@id tsp_lazy)
+# 
+# **Originally Contributed by**: Daniel Schermer
+# 
+# This notebook describes how to implement the Traveling Salesperson Problem in JuMP using solver-independent lazy constraints that dynamically separate subtours.
+# To be more precise, we use lazy constraints to cut off infeasible subtours only when necessary and not before needed.
+
+using JuMP
+import GLPK
+import LinearAlgebra
+import Random
+import Plots
+
+# # [Mathematical Formulation](@id tsp_model)
+# 
+# Assume that we are given a complete graph $\mathcal{G}(V,E)$ where $V$ is the set of vertices (or cities) and $E$ is the set of edges (or roads). 
+# For each pair of vertices $i, j \in V, i \neq j$ the edge $(i,j) \in E$ is associated with a weight (or distance) $d_{ij} \in \mathbb{R}^+$.
+# For this Notebook, we assume the problem to be symmetric, i.e., $d_{ij} = d_{ji} \, \forall i,j \in V$.
+
+# In the Traveling Salesperson Problem, we are tasked with finding a tour with minimal length that visits every vertex exactly once and then returns to the point of origin, i.e., a *hamiltonian cycle* with minimal weight.
+# 
+# In order to model the problem, we introduce a binary variable $x_{ij} \in \{0,1\} \; \forall i, j \in V$ that indicates if edge $(i,j)$ is part of the tour or not.
+# Using these variables, the Traveling Salesperson Problem can be modeled through the following Integer Linear Programming Formulation.
+# 
+# ## [Objective Function](@id tsp_objective)
+# The objective consists in minimizing the weighted edges (due to the assumed symmetry, the second sum only contains $i<j$).
+#
+# ```math
+# \text{min } \sum_{i \in V}  \sum_{j \in V, i < j} d_{ij} x_{ij}
+# ```
+#
+# Note that it is also psosible to use the following objective function instead.
+#
+# ```math
+# \text{min } \sum_{i \in V}  \sum_{j \in V} \dfrac{d_{ij} x_{ij}}{2}
+# ```
+#
+# ## [Constraints](@id tsp_constraints)
+# For the formulation of our problem, we need four classes of constraints which we will discuss in what follows.
+# Due to the presumed symmetry, the following constraints must hold.
+#
+# ```math
+# x_{ij} = x_{ji} \quad \forall i,j \in V
+# ```
+#
+# For each vertex $i$, exactly two edges must be selected that connect it to other vertices $j$ in the graph $G$.
+#
+# ```math
+# \sum_{j \in V} x_{ij} = 2 \quad \forall i \in V
+# ```
+# 
+# We do not permit loops to occur.
+#
+# ```math
+# x_{ii} = 0 \quad \forall i \in V
+# ```
+#
+# The number of vertices can be adjusted here.
```

Comment:
This needs a new heading to separate it from the math formulation

Reply:
Done, thanks for the suggestion. I moved the discussion of SEC to the model part to meaningfully separate the model from the implementation!